harvesters/src/main/java/org/fao/geonet/kernel/harvest/harvester/csw/Aligner.java

```diff
@@ -232,21 +223,44 @@ public HarvestResult align(Set<RecordInfo> records, List<HarvestError> errors) t
 
                 }
 
+                dataMan.forceIndexChanges();
                 result.totalMetadata++;
             } catch (Throwable t) {
                 errors.add(new HarvestError(this.context, t));
                 log.error("Unable to process record from csw (" + this.params.getName() + ")");
                 log.error("   Record failed: " +  ri.uuid + ". Error is: " + t.getMessage());
+				log.error(t);
             } finally {
                 result.originalMetadata++;
             }
         }
-        dataMan.forceIndexChanges();
+	}
 
-        log.debug("End of alignment for : " + params.getName());
+    /**
+     * Remove records no longer on the remote CSW server
+     * 
+     * @param records
+     * @throws Exception
+     */
+	@Transactional(value=TxType.REQUIRES_NEW)
+	public HarvestResult cleanupRemovedRecords(Set<String> records) throws Exception {
+		
+        if (cancelMonitor.get()) {
+            return result;
+        }
 
+        for (String uuid : localUuids.getUUIDs()) {
+            if (!records.contains(uuid)) {
+                String id = localUuids.getID(uuid);
+                log.debug("  - Removing old metadata with local id:" + id);
+                metadataManager.deleteMetadata(context, id);
+                result.locallyRemoved++;
+            }
+        }
+        dataMan.forceIndexChanges();
+        
         return result;
-    }
+	}
 
 
     private void addMetadata(RecordInfo ri, Integer ownerId, Integer groupId, String uuid) throws Exception {
```

Comment:
`ownerId` and `groupId` parameters are not used anymore.

Reply:
It looks like there was some weird merge so I also had to fix that on the metadata creation (was throwing null pointers if no user was assigned), but didn't check about the parameters.

Reply:
Fixed.